### PR TITLE
Refactor load medias functions

### PR DESCRIPTION
**Pull request by R-Jim**

### CODE REVIEW

File: frontend/src/pages/home/home.tsx [hunk 1]
```diff
@@ -77,73 +81,86 @@ export default class HomePage extends React.Component<HomePageProps, HomePageSta
...
             <div>
                 <div className="wrapper-overlay">
-                    {this.state.messageLoaded && this.state.announcementLoaded ? <ComboSection data={data}/> : <div/>}
+                    {this.state.messageLoaded && this.state.announcementLoaded ? <ComboSection data={data} /> : <div />}
                 </div>
             </div>
         )
     }
 
     compileCardData() {
```
Comment (zapplecat) [inline frontend/src/pages/home/home.tsx]: If things are async, would we run into any race conditions with compiling card data?

Comment (R-Jim) [reply frontend/src/pages/home/home.tsx]: It won't happen as long as these async functions are not updating the same state field

Comment (zapplecat) [reply frontend/src/pages/home/home.tsx]: What I mean is, if the functions are running async, will there be a possibility that this `compileCardData` runs before all the states have been fully populated? I'm not really privy to all the magics for frontends.

Comment (R-Jim) [reply frontend/src/pages/home/home.tsx]: ah, the component will re-render on every state change which `compileCardData ` used to compile card data. You can sure that what is rendered match the data in the state

**Review by DuongTyler — APPROVED** [review]
lgtm

**Pull request merged**